src/sentry/taskworker/registry.py

```diff
@@ -44,17 +44,12 @@ def __init__(
         self.default_expires = expires  # seconds
         self.default_processing_deadline_duration = processing_deadline_duration  # seconds
         self._registered_tasks: dict[str, Task[Any, Any]] = {}
-        self._producer: KafkaProducer | None = None
+        self._producer: SingletonProducer = SingletonProducer(self.producer, max_futures=1000)
 
-    @cached_property
     def producer(self) -> KafkaProducer:
```

Comment:
Should this become `_basic_producer`? We wouldn't want `TaskNamespace` or `Task` to use `.producer` as it doesn't handle the buffered futures or callbacks.